base/compiler/tfuncs.jl

```diff
@@ -1023,7 +1027,9 @@ function apply_type_tfunc(@nospecialize(headtypetype), @nospecialize args...)
             #        ai = ai.body
             #    end
             else
-                v = TypeVar(:_)
+                tvname = varnamectr <= length(_tvarnames) ? _tvarnames[varnamectr] : :_1
```

Comment:
`:_1` seems also potentially confusing, since thats also how our display code may mangle a name to produce a unique identifier

Reply:
`_` can't be used as input syntax for a typevar name, but ok. `_0`? Or what would you prefer?